recipe/meta.yaml

```diff
@@ -1,73 +1,51 @@
 {% set name = "xgboost" %}
-{% set version = "1.7.3" %}
+{% set version = "1.7.6" %}
+
+# note: r-xgboost is built from https://github.com/AnacondaRecipes/aggregateR/tree/R-4.3.1/r-xgboost-feedstock
 
 package:
   name: {{ name|lower }}-split
   version: {{ version }}
 
 source:
-  # we have to use git urls here to get the submodules needed for the
-  # build - the archive does not have these.
-  git_url: https://github.com/dmlc/xgboost
-  git_tag: v{{ version }}
+  url: https://github.com/dmlc/xgboost/releases/download/v1.7.6/xgboost.tar.gz
+  sha256: 0a54300dd274b98b7f039acffa006bec4875dace041fd9288422306fe7c379ca
   patches:
-    - 0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch            # [not win]
-    # Since the code is cloned from git, the line endings differ between unix and windows.
-    # Perhaps if git's autocrlf=false on the systems, one single patch would work.
-    # Not relying on that, this was the easiest way found to allow the patches to work on
-    # their respective platforms. Open to more thoughts.
-    - 0001-conda-Unbundle-libxgboost.-dll-dylib-so-WINDOWS.patch    # [win]
+    - 0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch
 
 build:
   skip: True  # [py<38]
   number: 0
 
 requirements:
   build:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - git                    # [not win]
-    - m2-git                 # [win]
     - patch                  # [not win]
     - m2-patch               # [win]
-    - cmake
-    - make  # [not win]
-    - llvm-openmp  # [osx]
-    - ninja
-  host:
-    - llvm-openmp  # [osx]
-    - python
-    - setuptools
-    - pip
-    - wheel
 
 outputs:
   - name: libxgboost
     script: install-libxgboost.sh  # [not win]
-    script: install-win-wrapper.bat  # [win]
+    script: install-libxgboost.bat  # [win]
     build:
       skip: True  # [py<38]
-      activate_in_script: True
-      ignore_run_exports:
-        - libgcc-ng  # [linux]
-        - libstdcxx-ng  # [linux]
       missing_dso_whitelist:
         - $RPATH/ld64.so.1  # [s390x]
-        - $RPATH/libgomp.so.1  # [linux]
-        - /usr/lib/libc++.1.dylib  # [osx]
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - git
         - cmake
-        - make         # [not win]
+        - ninja
         - llvm-openmp  # [osx]
       host:
         - llvm-openmp  # [osx]
       run:
         - llvm-openmp  # [osx]
         - _openmp_mutex  # [linux]
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libxgboost${SHLIB_EXT}       # [unix]
+        - if not exist %LIBRARY_BIN%\\xgboost.dll exit 1     # [win]
 
   - name: _py-xgboost-mutex
```

Comment:
what does this mutex do? afaiu it's not like conda will install two different versions of py-xgboost anyway?

Reply:
It doesn't do much, but I am inclined to revisit the day we had a gpu variant.